src/R/Editor/Impl/Data/RSessionDataObject.cs

```diff
@@ -108,22 +108,24 @@ public class RSessionDataObject : IRSessionDataObject {
                 return null;
             }
 
-            if (valueEvaluation.HasChildren) {
-                await TaskUtilities.SwitchToBackgroundThread();
-                REvaluationResultProperties properties =
-                    ExpressionProperty |
-                    AccessorKindProperty |
-                    TypeNameProperty |
-                    ClassesProperty |
-                    LengthProperty |
-                    SlotCountProperty |
-                    AttributeCountProperty |
-                    DimProperty |
-                    FlagsProperty |
-                    (RToolsSettings.Current.EvaluateActiveBindings ? ComputedValueProperty : 0);
-                var children = await valueEvaluation.DescribeChildrenAsync(properties, RValueRepresentations.Str(MaxReprLength), MaxChildrenCount);
-                return EvaluateChildren(children);
-            }
+            try {
+                if (valueEvaluation.HasChildren) {
+                    await TaskUtilities.SwitchToBackgroundThread();
```

Comment:
Let's move `await TaskUtilities.SwitchToBackgroundThread();` before try/catch block, otherwise `catch` will switch to UI thread.